### PR TITLE
Refactor to TypeScript and update messages builder

### PR DESCRIPTION
**Pull request by victorhmp**
#### What is the purpose of this pull request?

Refactor codebase to TypeScript and update builder for `react` and `messages`.

#### How should this be manually tested?

`cd react && yarn test`

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.
- [x] Refactor

### CODE REVIEW

File: messages/context.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{
+  "shippingEstimate-bd": "Shipping estimates in business days",
```
Comment (lucasecdb) [inline messages/context.json]: shouldn't the value be the key to the message id? e.g., in this case it should be
```
"shippingEstimate-bd": "shippingEstimate-bd"
```

**Review by lucasecdb — CHANGES_REQUESTED** [review]

File: messages/context.json (hunk 1 above)
Comment (victorhmp) [reply messages/context.json]: Wait, hahaha I don't think so. `context.json` should contain descriptions of what your message is for.

Comment (lucasecdb) [reply messages/context.json]: ah, ok then :-)

**Review by lucasecdb — APPROVED** [review]

File: messages/context.json [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+  "shippingEstimate-d": "Shipping estimates in days",
+  "shippingEstimate-h": "Shipping estimates in hours",
+  "shippingEstimate-m": "Shipping estimates in minutes",
+  "shippingEstimate-scheduled": "Shipping delivery window",
+  "shippingEstimate-scheduled-no-dates": "Scheduled",
+  "shippingEstimatePickup-bd": "Store pickup estimates in business days",
+  "shippingEstimatePickup-d": "Store pickup estimates in days",
+  "shippingEstimatePickup-h": "Store pickup estimates in hours",
+  "shippingEstimatePickup-m": "Store pickup estimates in minutes"
```
Comment (klzns) [inline messages/context.json]: I think all of those description might be "estimate".

Like: "Shipping estimate in hours", "Store pickup estimate in days", etc.

**Review by klzns — CHANGES_REQUESTED** [review]

File: messages/context.json (hunk 2 above)
Comment (victorhmp) [reply messages/context.json]: Yep! Makes sense hahahaha
Updated it :)

**Review by klzns — APPROVED** [review]

**Pull request merged**